gallery_dl/extractor/postmill.py

```diff
@@ -0,0 +1,204 @@
+                "instance": self.instance,
+            }
+
+            urls = []
+            if is_text_post or self.save_link_post_body:
+                urls.append((Message.Url, "text:" + body))
+
+            if is_image_post:
+                urls.append((Message.Url, url))
+            elif not is_text_post:
+                urls.append((Message.Queue, url))
+
+            data["count"] = len(urls)
+            yield Message.Directory, data
+            for data["num"], (msg, url) in enumerate(urls, 1):
+                if url.startswith("text:"):
+                    data["filename"], data["extension"] = "", "htm"
+                else:
+                    data = text.nameext_from_url(url, data)
+
+                yield msg, url, data
+
+
+class PostmillSubmissionsExtractor(PostmillExtractor):
+    """Base class for Postmill submissions extractors"""
+    whitelisted_parameters = ()
+
+    def __init__(self, match):
+        PostmillExtractor.__init__(self, match)
+        self.base = match.group(3)
+        self.sorting_path = match.group(4) or ""
+        self.query = {key: value for key, value in text.parse_query(
+            match.group(5) or "").items() if self.acceptable_query(key)}
+
+    def items(self):
+        url = self.root + self.base + self.sorting_path
+        if self.query:
+            url += "?" + urllib.parse.urlencode(self.query)
+
+        while url:
+            response = self.request(url)
+            if response.history:
+                redirect_url = response.url
+                if redirect_url == self.root + "/login":
+                    raise exception.StopExtraction(
+                        "HTTP redirect to login page (%s)", redirect_url)
+
+            for nav in text.extract_iter(response.text,
+                                         '<nav class="submission__nav">',
+                                         '</nav>'):
+                post_url = text.unescape(text.extr(nav, '<a href="', '"'))
+                yield Message.Queue, text.urljoin(url, post_url), \
+                    {"_extractor": PostmillPostExtractor}
+
+            url = text.unescape(text.extr(response.text,
+                                          '<link rel="next" href="', '">'))
+
+    def acceptable_query(self, key):
+        return key in self.whitelisted_parameters or key == "t" or \
+            (key.startswith("next[") and key.endswith("]"))
+
+
+BASE_PATTERN = PostmillExtractor.update({
+    "raddle": {
+        "root"   : None,
+        "pattern": (r"(?:raddle\.me|"
+                    r"c32zjeghcp5tj3kb72pltz56piei66drc63vkhn5yixiyk4cmerrjtid"
+                    r"\.onion)"),
+    }
+})
+SORTING_RE = r"(/(?:hot|new|active|top|controversial|most_commented))?"
+QUERY_RE = r"(?:\?([^#]+))?"
+
+
+class PostmillPostExtractor(PostmillExtractor):
+    """Extractor for a single submission URL"""
+    subcategory = "post"
+    pattern = BASE_PATTERN + r"/f/([\w\d_]+)/(\d+)(?:/.+)?$"
```

Comment:
```suggestion
    pattern = BASE_PATTERN + r"/f/(\w+)/(\d+)"
```
- `\w` includes both `\d` and `_`
- `.+` is almost always "wrong" is an RE (`[^/?#]+` is better for URLs)
- `(?:/.+)?$` can be omitted and tests still pass